pkg/object/dependson/strings.go

```diff
@@ -114,8 +114,8 @@ func ParseObjMetadata(objStr string) (object.ObjMetadata, error) {
 	fields := strings.Split(objStr, fieldSeparator)
 
 	if len(fields) != numFieldsClusterScoped && len(fields) != numFieldsNamespacedScoped {
-		return obj, fmt.Errorf("too many fields (expected %d or %d): %q",
-			numFieldsClusterScoped, numFieldsNamespacedScoped, objStr)
+		return obj, fmt.Errorf("expected %d or %d fields, found %d: %q",
```

Comment:
Is this a very helpful error message? Maybe more helpful to just print the value we tried to parse and then say that it couldn't be recognized as neither a cluster-scoped or namespace-scoped reference.

Reply:
Mostly i changed it because the message was wrong for the test i wrote that had no underscores. There were too few segments.

Both before and after the value itself is being appended.  For example, see https://github.com/kubernetes-sigs/cli-utils/pull/508/files#diff-84e8cf5b1fe1c9df4d3447f4414ddbcec8f820f7dbc0ee3cb56aaed1754f4d59R734-R737